com.unity.multiplayer.mlapi/Runtime/Configuration/NetworkConfig.cs

```diff
@@ -42,22 +42,41 @@ public class NetworkConfig
         public bool AllowRuntimeSceneChanges = false;
 
         /// <summary>
-        /// A list of spawnable prefabs
+        /// The previously set default player prefab
```

Comment:
Thanks for writing such a detailed comment.

In our UI/UX meeting with @morgane and Ross tomorrow I want to see what their thinking is, but if we have to go through lots of hoops I start to wonder if we should just go with not having a separate PlayerPrefab entry and just have an additional radio button next to each registered prefab to indicate whether you want it to be the player prefab.

Reply:
Well we have two meetings to really discuss.
There are three paths (ordered in personal preference):

1.  Provide the ability to not break existing v0.1.0 NetworkManager configurations as it pertains to the default PlayerPrefab.  Under this condition, you pointed out a unique scenario where the user had to delete the NetworkPrefab entry marked as Default Player before they could delete the PlayerPrefab setting in the editor (i.e. it would auto-populate back).  The PreviousPlayerPrefab is used to detect this specific condition.  I am sure there are other ways to do this.
2. Just break the NetworkManager configuration that would require anyone updating to populate the PlayerPrefab setting (i.e. no backwards-forwards compatibility)
3. Leave the additional field in NetworkPrefab list, which will be set once.  This path has never checked for more than one NetworkPrefab being marked as the "Player" prefab...so theoretically one could mark several entries but only one of the entries would apply.

Obviously I am a fan of #1, but we should most definitely discuss.

Reply:
Yeah, great summary...though on #3 I would (assuming it's possible) see if we could do radio buttons vs. checkboxes

Reply:
there are no radio buttons in Unity Editor inspector UI — we could reinvent it with checkboxes by keeping only one of them checked and others disabled.